input/otlp/logs.go

```diff
@@ -47,9 +47,18 @@ import (
 	"github.com/elastic/apm-data/model/modelpb"
 )
 
-func (c *Consumer) ConsumeLogs(ctx context.Context, logs plog.Logs) error {
+type ConsumeLogsResult struct {
+	RejectedLogRecords int64
+	AcceptedLogRecords int64
+}
+
+// ConsumeLogs consumes OpenTelemetry log data, converting into
+// the Elastic APM log model and sending to the reporter.
+// The returned ConsumeLogsResult contains the number of accepted and rejected log records.
+func (c *Consumer) ConsumeLogs(ctx context.Context, logs plog.Logs) (ConsumeLogsResult, error) {
+	totalCount := int64(logs.LogRecordCount()) // Use flattened log record count
 	if err := c.sem.Acquire(ctx, 1); err != nil {
-		return err
+		return ConsumeLogsResult{RejectedLogRecords: totalCount}, err
```

Comment:
I'd rather we disregard the result if there's a non-nil error returned. In which case, we should just return `ConsumeLogsResult{}, err`.